### PR TITLE
Added api

### PR DESCRIPTION
**Pull request by misikir21**
1. Use Axios with createAsyncThunk to make your HTTP requests to the backend
2. Fetch your books from the Bookstore API after the component has rendered
3. refactor  add book and remove book features to persist your changes in the server

### CODE REVIEW

File: src/components/route/addBook.jsx [hunk 1]
```diff
@@ -1,36 +1,37 @@
...
+
+  const data = {
+    item_id: uuidv4(),
+    title,
+    author,
+    category: 'Science-fiction',
+  };
+  const handleSubmit = () => {
+    dispatch(addBookToApi(data));
+    dispatch(addBook(data));
```
Comment (bienvenuushindi) [inline src/components/route/addBook.jsx:19]: :red_circle: [REQUIRED] I noticed that after clicking the add book button. The book  is rendered twice when a new book is added, Please 
Peek 2023-09-15 14-43
ensure that only one book is rendered per addition. :arrow_down:

**Review by bienvenuushindi — CHANGES_REQUESTED** [review]
## Hi @misikir21 ,

Good job so far!

## Highlights!
- GitFlow ✔️ 
- No linter errors ✔️ 
- Descriptive pull request ✔️ 
- Descriptive README ✔️ 

Hello ✋, you have done well reaching this point of the milestone 🤝 . But there are some issues that you still need to work on to go to the next project but you are almost there!

# Status: Required Changes ♻️⛔️
_Check the comments under the review and here 👇._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread and tag me `@julie-ify` if something is not 100% clear.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Comment by topeogunleye** [comment]
Hi @misikir21, 

While you made a great effort in this project, unfortunately, I cannot proceed to review your code.

## Invalid Code Review Request

You have submitted a project, that isn't runnning

image

Your Code Review Request will be marked as invalid in your Dashboard, so please submit a new one once you are ready 🙏

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag @topeogunleye in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

--------

_Invalid Code Review Request **does not count** into the code reviews limit._

**Review by topeogunleye — APPROVED** [review]
### Hi @misikir21, 

# Status: Approved ✔️
Excellent job on making the changes requested by the previous code reviewer. Keep it up 👏

Your project is complete! There is nothing else to say other than... it's time to merge it 💪
Congratulations! 🎉
congratulations
>
## To highlight :champagne: 
- Fixed the issue of rendering books twice. ✔️ 
 
## [Optional] suggestions:
* Nothing to mention.
 
 _Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._
 
Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag @topeogunleye in your question so I can receive the notification.



------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**